### PR TITLE
build: Include and format package.json and other projects

### PR DESCRIPTION
**Pull request by tylerbutler**
CI runs have a step that sets the version of packages, which edits the package.json and changes the whitespace formatting. We have a couple of different ways we set versions, so the quickest fix seemed to be to add steps to the "set package version" workflow that:

1. Runs a prettier check on just the package.json files in the release group BEFORE setting versions.
2. Runs a prettier format on just the package.json files in the release group AFTER setting versions.

Step 1 is needed because otherwise, badly formatted package.json files would go undetected, because we force them to be formatted in CI in step 2. The no extraneous files check already ignores package.json files because they change when the package version changes, so we can't rely on it as a backstop; we need to check them elsewhere in the pipeline.

An alternative to this workaround is to update our pipelines to use flub to set versions, and update flub to not touch whitespace. That will take more time, and this approach seems sufficient.

I also opportunistically updated the scripts and config for the benchmark project since it was outdated, and I updated the scripts for the release groups so that they lint and format the files in the root of the release group. And I updated the .prettierignore file as needed to exclude stuff.

Finally, there were some examples and PropertyDDS projects that did not have a config file, so I added one and formatted all the code.

### CODE REVIEW

**Comment by Josmithr** [comment]
> CI runs have a step that sets the version of packages, which edits the package.json and changes the whitespace formatting. We have a couple of different ways we set versions, so the quickest fix seemed to be to add steps to the "set package version" workflow that:
> 
> 1. Runs a prettier check on just the package.json files in the release group BEFORE setting versions.
> 2. Runs a prettier format on just the package.json files in the release group AFTER setting versions.
> 
> Step 1 is needed because otherwise, badly formatted package.json files would go undetected, because we force them to be formatted in CI in step 2. The no extraneous files check already ignores package.json files because they change when the package version changes, so we can't rely on it as a backstop; we need to check them elsewhere in the pipeline.
> 
> An alternative to this workaround is to update our pipelines to use flub to set versions, and update flub to not touch whitespace. That will take more time.

Do we need step 1 given that we run separate linter checks? PRs should catch if someone submits an unformatted package.json file, right? Seems like the job that does the modifications shouldn't need to worry about the formatting state prior to its edits.

File: tools/pipelines/templates/include-set-package-version.yml [hunk 1]
```diff
@@ -68,6 +73,23 @@ steps:
...
+  # them here because otherwise legitimate formatting problems would be hidden by the forced reformatting in CI.
+  - task: Bash@3
+    displayName: Check package.json formatting
+    inputs:
+      targetType: 'inline'
+      workingDirectory: ${{ parameters.buildDirectory }}
+      script: |
+        if [ -f "lerna.json" ]; then
+          # Run on all the release group packages, if in a release group
+          npx lerna exec prettier -- --check package.json
```
Comment (Josmithr) [inline tools/pipelines/templates/include-set-package-version.yml:87]: Is it safe for us to use `npx` for calls like this in pipelines? Don't we run the risk of downloading incompatible versions of lerna this way?

Comment (tylerbutler) [reply tools/pipelines/templates/include-set-package-version.yml:87]: Adding `--no-install` is a good idea. I'll do that.

**Comment by tylerbutler** [comment]
> Do we need step 1 given that we run separate linter checks? PRs should catch if someone submits an unformatted package.json file, right? Seems like the job that does the modifications shouldn't need to worry about the formatting state prior to its edits.

The problem is that setting the package version happens early in the CI flow, while linting happens at the end. So if we format the files as part of setting the version, then by the time they're linted they'll be formatted correctly. This is the flow:

1. Open a PR with spaces in a package.json.
2. PR build runs, and sets the package version as appropriate for the build.
3. The build then formats package.json using tabs.
4. The build then lints formatted code.
5. Dev merges, but the spaces are still there.

File: tools/pipelines/build-azure.yml [hunk 2]
```diff
@@ -98,6 +98,7 @@ extends:
     cgSubDirectory: packages
     checkoutSubmodules: true
     taskBundleAnalysis: false
+    usesPrettier: false
```
Comment (Josmithr) [inline tools/pipelines/build-azure.yml:101]: Shouldn't this one be true?

Comment (tylerbutler) [reply tools/pipelines/build-azure.yml]: yeah, fixed.

File: package.json [hunk 3]
```diff
@@ -39,15 +39,19 @@
...
     "layer-check": "fluid-layer-check --info build-tools/packages/build-tools/data/layerInfo.json",
     "lerna": "lerna",
-    "lint": "lerna run lint --no-sort --stream",
-    "lint:fix": "lerna run lint:fix --no-sort --stream",
+    "lint": "npm run prettier:root && lerna run lint --no-sort --stream",
+    "lint:fix": "npm run prettier:root:fix && lerna run lint:fix --no-sort --stream",
     "policy-check": "flub check policy --exclusions build-tools/packages/build-tools/data/exclusions.json",
     "policy-check:asserts": "flub check policy --exclusions build-tools/packages/build-tools/data/exclusions.json --handler assert-short-codes --fix",
     "policy-check:fix": "flub check policy --exclusions build-tools/packages/build-tools/data/exclusions.json --excludeHandler assert-short-codes --fix",
+    "prettier": "npm run prettier:root && lerna run prettier --no-sort --stream",
```
Comment (Josmithr) [inline package.json:51]: We probably need to add a prettier config to the repo root if we're adding root linting.

**Review by Josmithr — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (tylerbutler) [reply package.json:51]: Yeah, good catch. CI failed, but I can't figure out why it was working locally. \<shrug\>

**Review by Josmithr — APPROVED** [review]

**Pull request merged**